File: easiness-core/src/main/java/com/easiness/core/entities/DocumentItemEntity.java
package com.easiness.core.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.math.BigDecimal;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "document_item", indexes = @Index(name = "document_id_index", columnList = "document_id"))
public class DocumentItemEntity implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(name = "document_id")
    private Long documentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private ProductEntity product;

    @Column(name = "quantity")
    private BigDecimal quantity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "unit_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private UnitEntity unit;

    @Column(name = "cost_or_price")
    private BigDecimal costOrPrice;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "place_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private PlaceEntity place;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "affected_stock_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private StockEntity affectedStock;


}

File: easiness-ui/app/server/entity/document-item.entity.ts
import { Big } from "big.js";
import { Column, Entity, Index, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
import { BigColumn } from "../utils/decorators";
import { Document } from "./document.entity";
import { Place } from "./place.entity";
import { Product } from "./product.entity";
import { Stock } from "./stock.entity";

@Entity({ name: 'document_item' })
export class DocumentItem {
    @PrimaryGeneratedColumn()
    id: number

    @ManyToOne(() => Document, (document) => document.items)
    document: Document
    @Index()
    @Column()
    documentId: number

    @ManyToOne(() => Product)
    product: Product
    @Column()
    productId: number

    @BigColumn()
    quantity: Big

    @BigColumn()
    cost: Big

    @BigColumn(true)
    price!: Big

    @Column()
    unitId: number

    @ManyToOne(() => Place)
    place!: Place
    @Column({ nullable: true })
    placeId!: number

    @ManyToOne(() => Stock)
    affectedStock!: Stock
    @Column({ nullable: true })
    affectedStockId!: number

}